CMakeLists.txt

```diff
@@ -63,9 +63,23 @@ else()
     set(SFML_BUILD_EXAMPLES FALSE)
 endif()
 
+# add options for installing the examples
+if(SFML_BUILD_EXAMPLES)
+    sfml_set_option(SFML_INSTALL_EXAMPLES FALSE BOOL "TRUE to install the SFML examples, FALSE to leave them in the build directory")
+    sfml_set_option(SFML_INSTALL_EXAMPLES_PREFIX ${INSTALL_MISC_DIR} PATH "Install path prefix, prepended onto example install directory")
+endif()
+
 # add an option for building the API documentation
 sfml_set_option(SFML_BUILD_DOC FALSE BOOL "TRUE to generate the API documentation, FALSE to ignore it")
 
+# add option for installing the API documentation
+if(SFML_BUILD_DOC)
+    sfml_set_option(SFML_INSTALL_DOC FALSE BOOL "TRUE to install the SFML API documentation, FALSE to leave them in the build directory")
+endif()
+
+# add option for specifying path to install documentation files
+sfml_set_option(SFML_INSTALL_DOC_PREFIX ${INSTALL_MISC_DIR} PATH "Install path prefix, prepended onto the documentation install directory")
```

Comment:
> I don't like the solution too much

Yes, this is really not clear. You can disable building or installing the doc, but you still have to provide a valid install prefix for it :p

Do these two files really have to be part of the doc? I don't think it would hurt to just put them at the root of the development files install directory.